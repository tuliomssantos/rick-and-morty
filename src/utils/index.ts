export * from './createEmotionCache'
export * from './renderWithTeme'
