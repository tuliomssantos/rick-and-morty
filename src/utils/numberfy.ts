export interface INumberfy {
  (value: string | string[] | undefined, defaultValue: number): number
}

export const numberfy: INumberfy = (value, defaultValue) => {
  if (value) {
    return Number(value)
  }

  return defaultValue
}
