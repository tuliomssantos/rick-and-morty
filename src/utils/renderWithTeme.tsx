import { render, RenderResult } from '@testing-library/react'

import { CustomThemeProvider } from '../styles/theme'

import { QueryClient, QueryClientProvider } from 'react-query'

// import {setLogger} from 'react-query'
// setLogger({
//   log: console.log,
//   warn: console.warn,
//   // âœ… no more errors on the console
//   // eslint-disable-next-line @typescript-eslint/no-empty-function
//   error: () => {},
// })

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  const testQueryClient = createTestQueryClient()

  const { rerender, ...result } = render(
    <QueryClientProvider client={testQueryClient}>
      <CustomThemeProvider>{children}</CustomThemeProvider>
    </QueryClientProvider>
  )

  return {
    ...result,
    rerender: (rerenderChildren: React.ReactNode) =>
      rerender(
        <QueryClientProvider client={testQueryClient}>
          <CustomThemeProvider>{rerenderChildren}</CustomThemeProvider>
        </QueryClientProvider>
      ),
  }

  // return render(
  //   <QueryClientProvider client={queryClient}>
  //     <CustomThemeProvider>{children}</CustomThemeProvider>
  //   </QueryClientProvider>
  // )
}
