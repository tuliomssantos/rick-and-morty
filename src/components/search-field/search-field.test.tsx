/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-unused-vars */
import { renderWithTheme } from '@/src/utils/renderWithTeme'

import SearchField from '.'

const mockProps = {
  handleSearchChange: (name: string) => {},
  handleSearchLoading: () => {},
  searchLoading: false,
}

describe('<SearchField />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<SearchField {...mockProps} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
