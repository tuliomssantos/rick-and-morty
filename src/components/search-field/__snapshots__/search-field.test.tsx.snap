// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render correctly 1`] = `
<div
  class="css-0"
>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1wssyo0-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1iiaq29-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-vxvc7x-MuiInputBase-input-MuiOutlinedInput-input"
        id="input-with-icon-textfield"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-nnbavb"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
